Check for Shanghai in test node rather than Canyon

jyellick opened the pull request:
The test op-geth code adds withdrawals only if Canyon has been enabled. This worked upstream because Shanghai and Canyon were enabled concurrently, but it's not actually the right behavior.

souradeep-das commented on packages/message-relayer/src/service.ts:
```diff
@@ -239,7 +239,8 @@ export class MessageRelayerService extends BaseService<MessageRelayerOptions> {
             for (const cur of this.state.messageBuffer) {
-              const status = await this.state.messenger.getMessageStatus(cur)
+              const status =
+                await this.state.messenger.getMessageStatusFromContracts(cur)
```
since we only need succeded/failed status here, the finalization(state batch submission) log binary search can be avoided if needed. lmk if we want to avoid going through the search once more here

CAPtheorem replied:
Please pick whatever makes most sense

souradeep-das commented on packages/message-relayer/src/service.ts:
```diff
@@ -442,9 +444,10 @@ export class MessageRelayerService extends BaseService<MessageRelayerOptions> {
+              const status =
+                await this.state.messenger.getMessageStatusFromContracts(
+                  message
+                )
```
since we only need succeded/failed status here, the finalization(state batch submission) log binary search can be avoided if needed. lmk if we want to avoid going through the search once more here

CAPtheorem replied:
Please pick whatever makes most sense

jyellick committed:
Check for Shanghai in test node rather than Canyon

The test op-geth code adds withdrawals only if Canyon has been enabled.
This worked upstream because Shanghai and Canyon were enabled
concurrently, but it's not actually the right behavior.
